File: Source/Assemblies/Caching/Roblox.Caching/CacheInfo.ts
import { CacheabilitySettings } from './CacheabilitySettings';
import { ICacheInfo } from './ICacheInfo';

export class CacheInfo implements ICacheInfo {
	private _Cacheability: CacheabilitySettings = null;
	private _EntityType: string = null;

	public get Cacheability() {
		return this._Cacheability;
	}

	public get EntityType() {
		return this._EntityType;
	}

	public constructor(cacheability: CacheabilitySettings, entityType: string) {
		this._Cacheability = cacheability;
		this._EntityType = entityType;
	}
}

File: C#/CacheInfo.cs
using System;

namespace Roblox.Caching {
	public class CacheInfo : ICacheInfo {
		private	CacheabilitySettings _Cacheability = default(CacheabilitySettings);
		private string _EntityType = default(string);

		public CacheabilitySettings Cacheability {
			get { return _Cacheability; }
		}

		public string EntityType {
			get { return _EntityType; }
		}

		public CacheInfo(CacheabilitySettings cacheability, string entityType) {
			_Cacheability = cacheability;
			_EntityType = entityType;
		}
	}
}